tests/golden/xfstests

```diff
@@ -9,6 +9,8 @@ generic/011
 generic/013
 generic/014
 generic/020
+generic/023
```

Comment:
Let's put all the xfstests changes in one patch after the patches that add our support and our new tests.    (and then I guess update the commit message here from its first draft to just talk about rename2 support).